### PR TITLE
Logging fixes

### LINKED ISSUE #988: Provide a flag for log level
We still want verbose logging (both `INFO` and `ERROR` levels) by default, but if user's don't want logs to pile up (e.g. they don't care about debugging since they're just evaluating the tech) they should be able to provide a flag so that they can set the log level.

In this case, if they set it to `ERROR` they would probably get order(s) of magnitude less logs (most of ours are `INFO` level).

### PR DESCRIPTION
**Pull request by jdoliner**
fixes #988 and #987

### CODE REVIEW

**Comment by jdoliner** [comment]
@sjezewski whenever you have a chance.

File: src/server/pps/persist/persist.proto [hunk 1]
```diff
@@ -5,7 +5,7 @@ import "google/protobuf/timestamp.proto";
 import "client/pfs/pfs.proto";
 import "client/pps/pps.proto";
 
-package pachyderm.pps.persist;
+package pps.persist;
```
Comment (sjezewski) [inline src/server/pps/persist/persist.proto]: Nice catch. Every bit simpler helps

**Comment by sjezewski** [comment]
I notice that we removed all the log statements that report timing at the end of the request. What is the reason for that?

In #987 I understood the request to eliminate the redundancy between services `pachyderm.pfsserver.API` and `pfs.API` for instance (whose messages are otherwise redundant as far as I can tell other than that service name)

If you also eliminate the timing info you'll reduce the size by a factor of 4 instead of 2, so I could understand that desire.

If we are removing that in lieu of stats support for request timing (specifically the prometheus/grafana stats we haven't quite built yet), I'm ok w that ... but it does make me uncomfortable. I want to be clear about some utility we get from those logs.

I have found it useful to be able to see this information in the logs when debugging. Even if it was also reported in stats ... its really helpful to be able to correlate debug logs w certain statistics (bcz logs inevitably have more information about the nature of the request being made). Perhaps one alternative to meet this requirement (coordinating stats and logs) would be to report a unique request ID to both ... but normally stats services don't report individual requests that well?

But - as long as we agree we want to correlate them in the future, I'm good w this.

image

**Comment by jdoliner** [comment]
> I notice that we removed all the log statements that report timing at the end of the request. What is the reason for that?

I just removed the duplicates, timing info is still present in the logs it's just you'll only see 1 copy of those messages instead of 2. I agree with your sentiment that that's important info we don't want to lose. This PR shouldn't remove any information from the logs... except for the word `pachyderm` which we can hopefully all remember ourselves at this point :stuck_out_tongue:

**Pull request merged**